diff --git a/qemu/include/qemu/int128.h b/qemu/include/qemu/int128.h
index 0d0f2f1..f05a4fe 100644
--- a/qemu/include/qemu/int128.h
+++ b/qemu/include/qemu/int128.h
@@ -146,7 +146,7 @@ static inline Int128 bswap128(Int128 a)
 #else /* !CONFIG_INT128 */
 
 typedef struct Int128 Int128;
-#if !(defined(_MSC_VER) && defined(__clang__))
+#if !(defined(_MSC_VER) || defined(__SIZEOF_INT128__))
 typedef Int128 __int128_t;
 #endif
 
